plaidml/edsl/edsl.h

```diff
@@ -974,11 +981,16 @@ class gather {
     return *this;
   }
 
+  gather& batchDims(int batch_dims) {
+    batch_dims_ = Tensor(batch_dims);
+    return *this;
+  }
+
   ///
   /// Construct gather.
   ///
   Tensor build(edsl_source_location loc = edsl_source_location::current()) const {
-    std::vector<Tensor> args = {x_, y_, axis_, interpolation_mode_, nearest_mode_, cube_coeff_};
+    std::vector<Tensor> args = {x_, y_, axis_, interpolation_mode_, nearest_mode_, cube_coeff_, mode_, batch_dims_};
```

Comment:
Please update gather Python API in plaidml/edsl/__init__.py and check keras backend tests passed.

Reply:
Updated but only llvm_cpu target on keras backend passed. Both opecl and vulkan target failed. Is this expected?

Reply:
No, I think keras gather pass on gpu in December, not checked recently. Could you please check if it fails in plaidml-v1?

Reply:
Checked plaidml-v1 and failed with the same error on `intel_gen_ocl_spirv ` target:
```
plaidml.ffi.Error: vector::_M_range_check: __n (which is 0) >= this->size() (which is 0)
```
On `intel_gen` target:
```
LLVM ERROR: Can't add pass 'InlineLayers' restricted to 'func' on a PassManager intended to run on 'module', did you intend to nest?
```

Reply:
I see. The test seems is broken after recent updates. We do not need to worry about test fail on GPU in this PR, since it is not caused by us.